### PR TITLE
feat(operator): add logic to create services for inter-crg networking

### PR DESCRIPTION
**Pull request by strieflin**

### CODE REVIEW

File: klyshko-mp-spdz-cowgear/3RD-PARTY-LICENSES/sbom.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  {
+    "project": "https://github.com/llvm/llvm-project",
+    "license": "Apache-2.0"
+  },
+  {
+    "project": "https://github.com/openssl/openssl",
+    "license": "Apache-2.0"
+  },
+  {
+    "project": "https://github.com/osu-crypto/libOTe",
```
[the remark this answers is not in the document]
Comment (strieflin) [reply klyshko-mp-spdz-cowgear/3RD-PARTY-LICENSES/sbom.json:35]: Good catch. I have created the respective folder and added the license file to fulfill the license obligations.

File: klyshko-mp-spdz-cowgear/cowgear-offline.cpp [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        "Number of parties",  // Help description
+        "-N",                 // Short name
+        "--number-of-parties" // Long name
+    );
+    opt.add(
+        "",                                                  // Default value
+        1,                                                   // Required?
+        1,                                                   // Number of values expected
+        0,                                                   // Delimiter, if expecting multiple args
+        "This player's number, starting with 0 (required).", // Help description
```
[the remark this answers is not in the document]
Comment (strieflin) [reply klyshko-mp-spdz-cowgear/cowgear-offline.cpp:138]: Well, as of my understanding the original phrasing is correct English. Anyway, I changed it to "The number of this player".

File: klyshko-operator/controllers/tuplegenerationtask_controller.go [hunk 3]
```diff
@@ -256,8 +329,8 @@ func pvcName(key RosterEntryKey) string {
 	return key.Name + "-" + strconv.Itoa(int(key.PlayerID))
 }
 
-// createPVC creates a PVC used to transfer tuples between generator and provision pod for a task with the given key.
-func (r *TupleGenerationTaskReconciler) createPVC(ctx context.Context, key *RosterEntryKey) error {
+// getOrCreatePVC creates a PVC used to transfer tuples between generator and provision pod for a task with the given key.
+func (r *TupleGenerationTaskReconciler) getOrCreatePVC(ctx context.Context, key *RosterEntryKey) error {
```
[the remark this answers is not in the document]
Comment (strieflin) [reply klyshko-operator/controllers/tuplegenerationtask_controller.go:333]: The intention has been to express that the PVC either is returned directly when existing already or is created if not. However, I see that this semantics doesn't match the signature. Returning the PVC now in both cases.

File: klyshko-operator/controllers/tuplegenerationtask_controller.go [hunk 4]
```diff
@@ -523,3 +627,111 @@ func (r *TupleGenerationTaskReconciler) createGeneratorPod(ctx context.Context,
 	}
 	return pod, nil
 }
+
+func (r *TupleGenerationTaskReconciler) getOrCreateService(ctx context.Context, key *RosterEntryKey, task *klyshkov1alpha1.TupleGenerationTask) (*v1.Service, error) {
```
[the remark this answers is not in the document]
Comment (strieflin) [reply klyshko-operator/controllers/tuplegenerationtask_controller.go:631]: Yep. Added a comment to describe what the method does.

File: klyshko-mp-spdz-cowgear/cowgear-offline.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (sbckr) [reply klyshko-mp-spdz-cowgear/cowgear-offline.cpp:138]: You're right, obviously..

File: klyshko-mp-spdz-cowgear/kii-run.sh [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  done
+  return 0
+}
+
+# Setup offline executable command line arguments dictionary
+prime=$(cat /etc/kii/params/prime)
+declare -A argsByType=(
+  ["BIT_GFP"]="--field-type gfp --tuple-type bits --prime ${prime}"
+  ["BIT_GF2N"]="--field-type gf2n --tuple-type bits"
+  ["INPUT_MASK_GFP"]="--field-type gfp --tuple-type triples --prime ${prime}"
```
[the remark this answers is not in the document]
Comment (strieflin) [reply klyshko-mp-spdz-cowgear/kii-run.sh:54]: Yes, you are right. We discovered this issue a while ago already, but haven't addressed this so far. I have created #87 to track this. We should have a discussion on how and where (castor client, SDK, Klyshko) to address this.

**Review by sbckr — APPROVED** [review]
LGTM

**Comment by strieflin** [comment]
Just verified that everything is working as expected based on the SDK and the millionaires example.

**Pull request merged**